tools/src/main/java/org/apache/kafka/tools/EndToEndLatency.java

```diff
@@ -221,4 +303,165 @@ private static KafkaProducer<byte[], byte[]> createKafkaProducer(Optional<String
         producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer");
         return new KafkaProducer<>(producerProps);
     }
+
+    /**
+     * Converts legacy positional arguments to named arguments for backward compatibility.
+     *
+     * @param args the command line arguments to convert
+     * @return converted named arguments
+     * @throws Exception if the legacy arguments are invalid
+     * @deprecated Positional argument usage is deprecated and will be removed in Apache Kafka 5.0.
+     *             Use named arguments instead: --bootstrap-server, --topic, --num-records, --producer-acks, --record-size, --command-config
+     */
+    @Deprecated(since = "4.2", forRemoval = true)
+    static String[] convertLegacyArgsIfNeeded(String[] args) throws Exception {
+        if (args.length == 0) {
+            return args;
+        }
+
+        boolean hasNamedArgs = Arrays.stream(args).anyMatch(arg -> arg.startsWith("--"));
```

Comment:
`-` is legal character in topic names, so it could cause an issue if users create a topic like `topic--aaa`. Perhaps we should check the required `ops` instead?